webapp/src/components/github_milestone_selector/github_milestone_selector.jsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import IssueAttributeSelector from 'components/issue_attribute_selector';
+
+export default class GithubMilestoneSelector extends PureComponent {
+    static propTypes = {
+        repo: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        selectedMilestone: PropTypes.object.isRequired,
+        onChange: PropTypes.func.isRequired,
+        milestones: PropTypes.array.isRequired,
+        actions: PropTypes.shape({
+            getMilestones: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    loadMilestones = async () => {
+        if (this.props.repo === '') {
+            return [];
+        }
+
+        const milestones = await this.props.actions.getMilestones(this.props.repo);
+
+        if (!milestones || !milestones.data) {
+            return [];
+        }
+
+        return milestones.data.map((milestone) => ({
+            value: milestone.title,
+            label: milestone.title,
+        }));
+    };
+
+    onChange = (selection) => {
+        if (!selection || Object.keys(selection).length === 0) {
+            this.props.onChange({});
+            return;
+        }
+
+        // we have to find the selected milestone from the options in order to insert its number
+        const milestone = this.props.milestones.find((m) => m.title === selection.value);
+
+        this.props.onChange({
+            number: milestone.number,
+            label: milestone.title,
+            value: milestone.title,
+        });
+    }
+
+    render() {
+        const isSelectionBlank = Object.keys(this.props.selectedMilestone).length === 0;
+        const selection = isSelectionBlank ? '' : {
+            label: this.props.selectedMilestone.label,
+            value: this.props.selectedMilestone.value,
+        };
+
+        return (
+            <div className='form-group margin-bottom x3'>
+                <label className='control-label margin-bottom x2'>
+                    {'Milestone'}
+                </label>
+                <IssueAttributeSelector
```

Comment:
Once a milestone is selected, there is no option to remove it. 
![Screenshot from 2020-08-06 11-53-20](https://user-images.githubusercontent.com/16541325/89518642-79d47d80-d7db-11ea-8158-5d096b3f4ada.png)

For labels and assignees there is a `x` on the right side
![Screenshot from 2020-08-06 11-53-48](https://user-images.githubusercontent.com/16541325/89518710-97094c00-d7db-11ea-9100-cc22bc078e4b.png)

Reply:
The little `x` on the right side of Labels and Assignees is not gone. Was this change intentional? 

![Screenshot from 2020-08-07 09-52-23](https://user-images.githubusercontent.com/16541325/89622901-0b072b00-d894-11ea-90e4-aaf034ff9b65.png)

Reply:
@hanzei Why do we want to get rid of the `x` on the right side of labels & assignees? I think it's a good thing that the user is able to de-select a single label/assignee rather than having to clear all of the selected items. 

However, I realized that I removed the "clear all" `x` button on the far right side by mistake, so I'll add it back.

Reply:
Sorry, I had a typo in my comment. The `x` should in fact be there in the RHS.